### ISSUE TITLE
AdUnit defined native template

### ISSUE THREAD

**Issue by EskelCz** [prebid/Prebid.js#10006]
## Type of issue
Question

## Description
I'm trying to set up native ads, using adUnit defined templates, **without** a custom renderer
https://docs.prebid.org/prebid/native-implementation.html#42-implementing-an-adunit-defined-template
Using literally this example from the documentation:
https://docs.prebid.org/prebid/native-implementation.html#421-prebidjs-adunit-setup

I'm getting bids from magnite, which have all the native data in `native` key, on the bid response.
The issue is, it doesn't have an `ad` key, which means when I render it with `pbjs.renderAd`, this means that it fails with `Error trying to write ad.`, here: https://github.com/prebid/Prebid.js/blob/5199712a3f8534a5ae9ef68203fbbbe86eb1387a/src/prebid.js#L583

I don't see where the native template rendering should kick in for this scenario. My template is there (in the mediaTypes.native), native data on the bidResponse is there, but renderAd doesn't seem to be made for that purpose. What am I missing?

Thanks

### Expected results
Native ad being rendered

### Actual results
MESSAGE: Calling renderAd with adId :7e523379a1127f
ERROR: Error trying to write ad. No ad for bid response id: 7e523379a1127f

## Platform details
Mac OS, Chrome

## Other information
Prebid 7.39.0

**Comment by EskelCz** [comment]
@dgirardi I see, thanks for the clarification. We'd rather not use the PUC, so I'll watch the proposal. Any idea on the timeframe? I'm wondering whether to do the bidsBackHandler, figure out how to trigger the event of the ad being rendered (to trigger analytics adapters etc) and rework it once it becomes a feature... or just wait for it for a few weeks. :)

**Comment by dgirardi** [comment]
@EskelCz I can't make any promises, but my best guess is in the order of a few months.

**Comment by EskelCz** [comment]
@dgirardi One more question if you don't mind, so I don't have to open a new issue for a similar thing:

For the direct/custom native rendering I used this postMessage, which works well:
`{ message: 'Prebid Native', adId: winningBid.adId }`

I ran into another use case for a custom rendering (rendering without an iframe), this time of a regular *banner* and I can't figure out how to trigger the same adRenderSucceded event in prebid.
I tried these messages:
```
{ message: 'Prebid Native', adId: winningBid.adId } -- fails because regular banner doesn't have .native
{ message: 'Prebid Banner', adId: winningBid.adId } -- nothing, doesn't trigger analytics
{ adId: winningBid.adId } -- the same thing, doesn't work
```
Is there some other message that would work?
Thanks a lot.
